### PR TITLE
Template Editing Mode: Fix options dropdown

### LINKED ISSUE #37427: Template Editor toolbar dropdown broken
### Description

When editing a template via the template editing mode in the post editor, the dropdown in the toolbar appears to be broken.

### Step-by-step reproduction instructions

1. Use a block theme or theme supporting template editing mode
2. Head to a post or page
3. Under Templates in sidebar > Edit
4. Try to click the dropdown in the toolbar.

### Screenshots, screen recording, code snippet

https://user-images.githubusercontent.com/26996883/146278794-9e1090d8-0a39-4cf7-91c6-e3e288d43c75.mov

Screen Shot 2021-12-15 at 6 06 43 PM
Screen Shot 2021-12-15 at 6 07 57 PM

### Environment info

- WordPress 5.9 beta 3
- No GB
- TT2 

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by Mamaduka**
## Description
PR makes the following changes in the Template Title component:

* Remove template descriptions. Templates visible in the post/page dropdown currently don't support this feature.
* Conditionally hides dropdown when the template has no actionable options - change the title or delete.

P.S. We should change the "Delete" text to "Clear customizations" when a template has a theme file. I will address that in separate PR since we're in string freeze, and I want this to be backported in beta 4.

Resolves #37427.

## How has this been tested?
Using TT2

1. Create/edit a page.
2. Create a new template.
3. Make sure the "Template Options" dropdown is visible and works.
4. Switch to custom theme templates and see that only template title is displayed.

## Screenshots 
| with options | without options |
| --- | --- |
|CleanShot 2021-12-16 at 12 10 34|CleanShot 2021-12-16 at 12 10 52|

## Types of changes
Bugfix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by jameskoster** [comment]
> Conditionally hides dropdown when the template has no actionable options - change the title or delete.

I don't know that we necessarily have to hide the dropdown here. For example it may be useful to view the description to better understand the currently-applied template – especially as some of them have somewhat ambiguous names.

Here's the dropdown when editing the Page template in the site editor:

Screenshot 2021-12-16 at 10 52 30

I'm inclined to say we should work towards consistency across editing contexts. IE display the dropdown, without any actions.

**Comment by Mamaduka** [comment]
Thanks for catching that. So only hide dropdown when we have nothing to display, like custom templates by theme.

**Comment by jameskoster** [comment]
Yes I think so. Although it may be worth building in a condition to check for a description and display the dropdown when one exists. It seems very likely that it will be possible for custom theme templates to include descriptions in the future.

**Comment by Mamaduka** [comment]
I fixed the logic and added back descriptions.

**Comment by jameskoster** [comment]
I'm only seeing the description.

Screenshot 2021-12-16 at 12 54 44

Do you think it makes sense to include the title like in the screenshot from the Site Editor above?

**Comment by Mamaduka** [comment]
> Do you think it makes sense to include the title like in the screenshot from the Site Editor above?

Sure. I'm all for consistency.

**Comment by Mamaduka** [comment]
@jameskoster started working on this, and I changed my mind 😅  I think displaying the Template title twice isn't necessary.

**Comment by jameskoster** [comment]
Hah, I don't have a strong feeling but would be inclined to say we should match the Site Editor for now, and then explore this particular detail in a separate issue. I suspect we'll revise this popover soon, possibly as a part of https://github.com/WordPress/gutenberg/issues/36951.

**Comment by Mamaduka** [comment]
@jameskoster sounds good. I will update the component tomorrow 👍

File: packages/edit-post/src/components/header/template-title/edit-template-title.js [hunk 1]
```diff
@@ -29,6 +29,10 @@ export default function EditTemplateTitle() {
 	const { getEditorSettings } = useSelect( editorStore );
 	const { updateEditorSettings } = useDispatch( editorStore );
 
+	if ( template.has_theme_file ) {
```
Comment (ntsekouras) [inline packages/edit-post/src/components/header/template-title/edit-template-title.js:32]: Why we need this?

Comment (Mamaduka) [reply packages/edit-post/src/components/header/template-title/edit-template-title.js:32]: Currently, only user-created Templates can be renamed.

**Review by ntsekouras — COMMENTED** [review]
Thanks George! Code wise this LGTM.

>@jameskoster sounds good. I will update the component tomorrow 👍

Is there anything left to decided/changed?

**Comment by jameskoster** [comment]
> Is there anything left to decided/changed?

Yup we need to display the title in the popover to match the site editor.

**Comment by Mamaduka** [comment]
Updated the design.

CleanShot 2021-12-18 at 17 22 33

**Comment by paaljoachim** [comment]
Some brainstorming based on what I read above. (It might be a sidetrack...:)

Template editing mode should be present in one way or another be it in the Site Editor and (somewhat similar much more limited in the) Post Editor. 
Keeping a consistency between editing. At the moment we do not have the [Post/Page Title in the top bar](https://github.com/WordPress/gutenberg/issues/27093#issuecomment-946607869) in the Post Editor. (It should be included soon to expand on the possibilities for design/layout in the Post Editor.)

Let's say that a template area does not have any areas included. It would be helpful to keep the same structure in place.
As it would give the user the same information as seen in other templates but it would then naturally give other information within the same UI.

No-areas-included


Now let's say that we also need to remove the "Browse all templates" bottom link.

Here is the link removed but the bottom black area is still there.
Remove bottom template link

Of course it would be better to actually mention why the link is removed instead of just removing the link like I did.

**Comment by jameskoster** [comment]
I updated the styling to match the site editor more closely. This affected the changes in https://github.com/WordPress/gutenberg/pull/37450/ so I brought those over here as well.

Screenshot 2021-12-20 at 11 15 31
Screenshot 2021-12-20 at 11 15 44

**Review by jameskoster — APPROVED** [review]

**Review by carolinan — APPROVED** [review]

**Pull request merged**